docs/reference/sql/syntax/lexical-structure.md

```diff
@@ -0,0 +1,201 @@
+---
+layout: page
+title: Lexical structure data
+tagline: Structure of SQL commands and statements in ksqlDB 
+description: Details about SQL commands and statements in ksqlDB 
+keywords: ksqldb, sql, keyword, identifier, constant, operator
+---
+
+SQL is a domain-specific language for managing and manipulating data. It’s
+used primarily to work with structured data, where the types and relationships
+across entities are well-defined. Originally adopted for relational databases,
+SQL is rapidly becoming the language of choice for stream processing. It’s
+declarative, expressive, and ubiquitous.
+
+The American National Standards Institute (ANSI) maintains a standard for the
+specification of SQL. SQL-92, the third revision to the standard, is generally
+the most recognized form of the specification. Beyond the standard, there are
+many flavors and extensions to SQL so that it can express programs beyond
+what's possible with the SQL-92 grammar.
+
+ksqlDB’s SQL grammar was built  initially around Presto's grammar and has been
+extended judiciously. ksqlDB goes beyond SQL-92, because the standard currently
+has no constructs for streaming queries, which are a core aspect of this project.
+
+## Syntax
+
+SQL inputs are made up of a series of statements. Each statements is made up of
+a series of tokens and ends in a semicolon (`;`). The tokens that apply depend
+on the statement being invoked.
+
+A token is any keyword, identifier, backticked identifier, literal, or special
+character. By convention, tokens are separated by whitespace, unless there is
+no ambiguity in the grammar. This happens when tokens flank a special character.
+
+The following example statements are syntactically valid ksqlDB SQL input:
+
+```sql
+INSERT INTO s1 (a, b) VALUES ('k1', 'v1');
+
+CREATE STREAM s2 AS
+    SELECT a, b
+    FROM s1
+    EMIT CHANGES;
+
+SELECT * FROM t1 WHERE k1='foo' EMIT CHANGES;
+```
+
+## Keywords
+
+Some tokens, such as `SELECT`, `INSERT`, and `CREATE`, are _keywords_.
+Keywords are reserved tokens that have a specific meaning in ksqlDB's syntax.
+They control their surrounding allowable tokens and execution semantics.
+Keywords are case insensitive, meaning `SELECT` and `select` are equivalent.
+You can't create an identifier that is already a keyword, unless you use
+backticked identifiers.
```

Comment:
not all keywords are reserved - I believe all of the following can be used without escaping them:
```
nonReserved
    : SHOW | TABLES | COLUMNS | COLUMN | PARTITIONS | FUNCTIONS | FUNCTION | SESSION
    | STRUCT | MAP | ARRAY | PARTITION
    | INTEGER | DATE | TIME | TIMESTAMP | INTERVAL | ZONE
    | YEAR | MONTH | DAY | HOUR | MINUTE | SECOND
    | EXPLAIN | ANALYZE | TYPE | TYPES
    | SET | RESET
    | IF
    | SOURCE | SINK
    | PRIMARY | KEY
    | EMIT
    | CHANGES
    | FINAL
    | ESCAPE
    | REPLACE
    | ASSERT
    ;
```